program/src/instructions/token22/close_expired_listing.rs

```diff
@@ -28,6 +28,7 @@ pub struct CloseExpiredListingT22<'info> {
         bump = list_state.bump[0],
         close = rent_destination,
         has_one = owner,
```

Comment:
I think we just remove the list_state.asset_id == asset.key() in the other handler. Otherwise we have inconsistencies + too many redundancies.